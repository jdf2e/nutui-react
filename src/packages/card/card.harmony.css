.nut-card {
  width: 100%;
  display: flex;
  background-color: inherit;
  border-radius: 4px;
}
.nut-card-left {
  width: 120px;
  height: 120px;
  flex-shrink: 0;
}
.nut-card-left > img {
  display: block;
  width: 100%;
  height: 100%;
  border-radius: 4px;
}
.nut-card-right {
  flex: 1;
  padding: 0 10px 8px;
}
.nut-card-right-title {
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 2;
  overflow: hidden;
  word-break: break-all;
  line-height: 1.5;
  font-size: 14px;
  color: #1a1a1a;
}
.nut-card-right-price {
  display: flex;
  align-items: center;
  height: 18px;
  line-height: 18px;
  margin-top: 9px;
}
.nut-card-right-price .nut-price .nut-price-big {
  font-size: 18px;
}
.nut-card-right-price .nut-price .nut-price-symbol,
.nut-card-right-price .nut-price .nut-price-point,
.nut-card-right-price .nut-price .nut-price-small {
  font-size: 12px;
}
.nut-card-right-price-origin.nut-price {
  margin-left: 2px;
  color: #d2a448;
}
.nut-card-right-price-origin.nut-price .nut-price-big,
.nut-card-right-price-origin.nut-price .nut-price-symbol,
.nut-card-right-price-origin.nut-price .nut-price-point,
.nut-card-right-price-origin.nut-price .nut-price-small {
  font-size: 12px;
}
.nut-card-right-other {
  display: flex;
  align-items: center;
  padding: 5px 0 2px;
}
.nut-card-right-other .nut-tag {
  padding: 0 2px;
  margin-right: 5px;
  font-size: 10px;
}
.nut-card-right-shop {
  display: flex;
  justify-content: space-between;
  align-items: center;
}
.nut-card-right-shop-name {
  line-height: 1.5;
  color: #595959;
  font-size: 12px;
  padding-top: 4px;
}

[dir=rtl] .nut-card-right-price-origin.nut-price,
.nut-rtl .nut-card-right-price-origin.nut-price {
  margin-left: 0;
  margin-right: 2px;
}
[dir=rtl] .nut-card-right-other .nut-tag,
.nut-rtl .nut-card-right-other .nut-tag {
  margin-right: 0;
  margin-left: 5px;
}