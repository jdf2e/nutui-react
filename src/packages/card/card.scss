.nut-card {
  width: 100%;
  display: flex;
  .nut-card__left {
    width: 120px;
    height: 120px;
    flex-shrink: 0;
    > img {
      display: block;
      width: 100%;
      height: 100%;
    }
  }
  .nut-card__right {
    flex: 1;
    padding: 0 10px 8px;
    .nut-card__right__title {
      display: -webkit-box;
      -webkit-box-orient: vertical;
      -webkit-line-clamp: 2;
      overflow: hidden;
      word-break: break-all;
      line-height: 1.5;
      font-size: 14px;
    }
    .nut-card__right__price {
      display: flex;
      align-items: center;
      height: 18px;
      line-height: 18px;
      margin-top: 9px;
      .nut-price {
        .nut-price__symbol {
          font-size: 12px;
        }
        .nut-price__big {
          font-size: 18px;
        }
        .nut-price__point,
        .nut-price__small {
          font-size: 12px;
        }
      }
      .nut-card__right__price__origin {
        &.nut-price {
          margin-left: 2px;
          color: #d2a448;
          .nut-price__symbol {
            font-size: 12px;
          }
          .nut-price__big {
            font-size: 12px;
          }
          .nut-price__point,
          .nut-price__small {
            font-size: 12px;
          }
        }
      }
    }
    .nut-card__right__other {
      display: flex;
      align-items: center;
      padding: 5px 0 2px;
      .nut-tag {
        padding: 0 2px;
        margin-right: 5px;
        font-size: $font-size-0;
      }
    }
    .nut-card__right__shop {
      display: flex;
      justify-content: space-between;
      align-items: center;
      .nut-card__right__shop__name {
        line-height: 1.5;
        color: #999;
        font-size: 12px;
        padding-top: 4px;
      }
    }
  }
}
