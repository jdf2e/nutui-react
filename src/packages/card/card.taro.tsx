import React, { FunctionComponent } from 'react'
import classNames from 'classnames'
import Price from '@/packages/price/index.taro'
import Tag from '@/packages/tag/index.taro'
import { BasicComponent, ComponentDefaults } from '@/utils/typings'

export interface CardProps extends BasicComponent {
  src: string
  title: string
  price: string
  vipPrice: string
  shopDescription: string
  delivery: string
  shopName: string
  description: React.ReactNode
  tag: React.ReactNode
  priceTag: React.ReactNode
  extra: React.ReactNode
}
const defaultProps = {
  ...ComponentDefaults,
  src: '',
  title: '',
  price: '',
  vipPrice: '',
  shopDescription: '',
  delivery: '',
  shopName: '',
  description: '',
  tag: '',
  priceTag: '',
  extra: '',
} as CardProps
export const Card: FunctionComponent<
  Partial<CardProps> & React.HTMLAttributes<HTMLDivElement>
> = (props) => {
  const {
    className,
    style,
    src,
    title,
    price,
    vipPrice,
    shopDescription,
    delivery,
    shopName,
    tag,
    priceTag,
    description,
    extra,
    ...rest
  } = {
    ...defaultProps,
    ...props,
  }
  const classPrefix = 'nut-card'
  return (
    <div className={classNames(classPrefix, className)} style={style} {...rest}>
      <div className={`${classPrefix}-left`}>
        <img src={src} alt="" />
      </div>
      <div className={`${classPrefix}-right`}>
        <div className={`${classPrefix}-right-title`}>{title}</div>
        {description}
        <div className={`${classPrefix}-right-price`}>
          <Price size="normal" price={price} />
          {priceTag || (
            <Price
              size="normal"
              className={`${classPrefix}-right-price-origin`}
              price={vipPrice}
            />
          )}
        </div>
        <div className={`${classPrefix}-right-other`}>
          {tag || (
            <>
              <Tag type="danger">{shopDescription}</Tag>
              <Tag plain>{delivery}</Tag>
            </>
          )}
        </div>
        <div className={`${classPrefix}-right-shop`}>
          <div className={`${classPrefix}-right-shop-name`}>{shopName}</div>
          {extra}
        </div>
      </div>
    </div>
  )
}

Card.displayName = 'NutCard'
