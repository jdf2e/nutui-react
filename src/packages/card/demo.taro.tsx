import React from 'react'
import Taro from '@tarojs/taro'
import { useTranslate } from '@/sites/assets/locale/taro'
import Header from '@/sites/components/header'

import Demo1 from './demos/taro/demo1'
import Demo2 from './demos/taro/demo2'
import Demo3 from './demos/taro/demo3'
import Demo4 from './demos/taro/demo4'
import Demo5 from './demos/taro/demo5'

const CardDemo = () => {
  const [translated] = useTranslate({
    'zh-CN': {
      basic: '基础用法',
      customProduct: '自定义商品标签',
      customShop: '自定义店铺介绍',
      customFooter: '自定义右下角内容',
    },
    'zh-TW': {
      basic: '基础用法',
      customProduct: '自定義商品標簽',
      customShop: '自定義店鋪介紹',
      customFooter: '自定義右下角內容',
    },
    'en-US': {
      basic: 'Basic Usage',
      customProduct: 'Custom prolist',
      customShop: 'Custom Content',
      customFooter: 'Customize bottom right content',
    },
  })
  return (
    <>
      <Header />
      <div className={`demo ${Taro.getEnv() === 'WEB' ? 'web' : ''}`}>
        <h2>{translated.basic}</h2>
        <Demo1 />
        <h2>{translated.customProduct}</h2>
        <Demo2 />
        <h2>{translated.customProduct}</h2>
        <Demo3 />
        <h2>{translated.customShop}</h2>
        <Demo4 />
        <h2>{translated.customFooter}</h2>
        <Demo5 />
      </div>
    </>
  )
}

export default CardDemo
