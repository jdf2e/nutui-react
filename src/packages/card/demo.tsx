import React from 'react'
import { useTranslate } from '../../sites/assets/locale'

import Demo1 from './demos/h5/demo1'
import Demo2 from './demos/h5/demo2'
import Demo3 from './demos/h5/demo3'
import Demo4 from './demos/h5/demo4'
import Demo5 from './demos/h5/demo5'

const CardDemo = () => {
  const [translated] = useTranslate({
    'zh-CN': {
      basic: '基础用法',
      customProduct: '自定义商品标签',
      customShop: '自定义店铺介绍',
      customPriceIcon: '价格后自定义标签',
      customFooter: '自定义右下角内容',
    },
    'zh-TW': {
      basic: '基础用法',
      customProduct: '自定義商品標簽',
      customShop: '自定義店鋪介紹',
      customPriceIcon: '價格後自定義標簽',
      customFooter: '自定義右下角內容',
    },
    'en-US': {
      basic: 'Basic Usage',
      customProduct: 'Custom prolist',
      customShop: 'Custom Content',
      customPriceIcon: 'Price after custom tag',
      customFooter: 'Customize bottom right content',
    },
  })
  return (
    <>
      <div className="demo">
        <h2>{translated.basic}</h2>
        <Demo1 />
        <h2>{translated.customProduct}</h2>
        <Demo2 />
        <h2>{translated.customPriceIcon}</h2>
        <Demo3 />
        <h2>{translated.customShop}</h2>
        <Demo4 />
        <h2>{translated.customFooter}</h2>
        <Demo5 />
      </div>
    </>
  )
}

export default CardDemo
