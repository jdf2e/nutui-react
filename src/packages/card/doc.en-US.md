# Card

## Intro

Used to display product pictures, prices and other information.

## Install

```tsx
import { Card, Price, Tag } from '@nutui/nutui-react'
```

## Demo

### Basic Usage

:::demo

```tsx
import React from 'react'
import { Card, Price, Tag } from '@nutui/nutui-react'

const App = () => {
  const state = {
    src: '//img10.360buyimg.com/n2/s240x240_jfs/t1/210890/22/4728/163829/6163a590Eb7c6f4b5/6390526d49791cb9.jpg!q70.jpg',
    title: 'title',
    price: '388',
    vipPrice: '378',
    shopDescription: 'description',
    delivery: 'delivery',
    shopName: 'shopName>',
  }
  return (
    <Card
      src={state.src}
      title={state.title}
      price={state.price}
      vipPrice={state.vipPrice}
      shopDescription={state.shopDescription}
      delivery={state.delivery}
      shopName={state.shopName}
     />
  )
}
export default App
```

:::

### Custom prolist

:::demo

```tsx
import React from 'react'
import { Card, Price, Tag } from '@nutui/nutui-react'

const App = () => {
  const state = {
    src: '//img10.360buyimg.com/n2/s240x240_jfs/t1/210890/22/4728/163829/6163a590Eb7c6f4b5/6390526d49791cb9.jpg!q70.jpg',
    title: 'title',
    price: '388',
    vipPrice: '378',
    shopDescription: 'description',
    delivery: 'delivery',
    shopName: 'shopName>',
  }
  const wordStyles = {
    padding: '0 5px',
    borderRadius: '1px',
    fontSize: '10px',
    height: '15px',
    lineHeight: '15px',
    color: '#999',
    backgroundColor: '#f2f2f7',
    marginRight: '5px',
  }
  return (
    <Card
      src={state.src}
      title={state.title}
      price={state.price}
      vipPrice={state.vipPrice}
      shopDescription={state.shopDescription}
      delivery={state.delivery}
      shopName={state.shopName}
      description={
        <div
          className="search_prolist_attr"
          style={{
            display: 'inline-flex',
            margin: '3px 0 1px',
            height: '15px',
          }}
        >
          {['tag', 'tag', 'tag'].map((item) => {
            return (
              <span style={wordStyles} className="word" key={item}>
                {item}
              </span>
            )
          })}
        </div>
      }
     />
  )
}
export default App
```

:::

### Price after custom tag

:::demo

```tsx
import React from 'react'
import { Card, Price, Tag } from '@nutui/nutui-react'

const App = () => {
  const state = {
    src: '//img10.360buyimg.com/n2/s240x240_jfs/t1/210890/22/4728/163829/6163a590Eb7c6f4b5/6390526d49791cb9.jpg!q70.jpg',
    title: 'title',
    price: '388',
    vipPrice: '378',
    shopDescription: 'description',
    delivery: 'delivery',
    shopName: 'shopName>',
  }
  const tagStyles = {
    display: 'inline-block',
    verticalAlign: 'middle',
    marginRight: '5px',
    marginLeft: '2px',
    height: '14px',
  }
  return (
    <Card
      src={state.src}
      title={state.title}
      price={state.price}
      vipPrice={state.vipPrice}
      shopDescription={state.shopDescription}
      delivery={state.delivery}
      shopName={state.shopName}
      priceTag={
        <img
          style={tagStyles}
          src="https://img11.360buyimg.com/jdphoto/s58x28_jfs/t9451/359/415622649/15318/b0943e5d/59a78495N3bd2a9f8.png"
          alt=""
        />
      }
     />
  )
}
export default App
```

:::

### Custom Content

:::demo

```tsx
import React from 'react'
import { Card, Price, Tag } from '@nutui/nutui-react'

const App = () => {
  const state = {
    src: '//img10.360buyimg.com/n2/s240x240_jfs/t1/210890/22/4728/163829/6163a590Eb7c6f4b5/6390526d49791cb9.jpg!q70.jpg',
    title: 'title',
    price: '388',
    vipPrice: '378',
    shopDescription: 'description',
    delivery: 'delivery',
    shopName: 'shopName>',
  }
  return (
    <Card
      src={state.src}
      title={state.title}
      price={state.price}
      vipPrice={state.vipPrice}
      shopDescription={state.shopDescription}
      delivery={state.delivery}
      shopName={state.shopName}
      tag={<div>Custom Content</div>}
     />
  )
}
export default App
```

:::

### Customize bottom right content

:::demo

```tsx
import React from 'react'
import { Card, Price, Tag } from '@nutui/nutui-react'

const App = () => {
  const state = {
    src: '//img10.360buyimg.com/n2/s240x240_jfs/t1/210890/22/4728/163829/6163a590Eb7c6f4b5/6390526d49791cb9.jpg!q70.jpg',
    title: 'title',
    price: '388',
    vipPrice: '378',
    shopDescription: 'description',
    delivery: 'delivery',
    shopName: 'shopName>',
  }

  return (
    <Card
      src={state.src}
      title={state.title}
      price={state.price}
      vipPrice={state.vipPrice}
      shopDescription={state.shopDescription}
      delivery={state.delivery}
      shopName={state.shopName}
      extra={<div style={{ fontSize: '12px' }}>custom</div>}
     />
  )
}
export default App
```

:::

## Card

### Props

| Property | Description | Type | Default |
| --- | --- | --- | --- |
| src | Left thumb image | `string` | `-` |
| title | Title | `string` | `-` |
| price | Price | `string` | `-` |
| vipPrice | vip-price | `string` | `-` |
| shopDescription | shop-description | `string` | `-` |
| delivery | delivery | `string` | `-` |
| shopName | shop-name | `string` | `-` |
| description | Custom product introduction | `ReactNode` | `-` |
| priceTag | Custom content behind the price | `ReactNode` | `-` |
| tag | Custom shop introduction | `ReactNode` | `-` |
| extra | Customize bottom right content | `ReactNode` | `-` |

## Theming

### CSS Variables

The component provides the following CSS variables, which can be used to customize styles. Please refer to [ConfigProvider component](#/en-US/component/configprovider).

| Name | Description | Default |
| --- | --- | --- |
| \--nutui-card-border-radius | The size of the rounded corners of the card picture and card | `4px` |
