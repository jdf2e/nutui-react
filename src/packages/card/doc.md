# Card 商品卡片

## 介绍

商品卡片，用于展示商品的图片、价格等信息

## 安装

```js
// react
import { Card, Price, Tag } from '@nutui/nutui-react';
```

## 代码演示

### 基本用法

:::demo

```ts
import React from "react";
import { Card, Price, Tag } from '@nutui/nutui-react';

const App = () => {
  const state = {
    imgUrl:
      '//img10.360buyimg.com/n2/s240x240_jfs/t1/210890/22/4728/163829/6163a590Eb7c6f4b5/6390526d49791cb9.jpg!q70.jpg',
    title: '【活蟹】湖塘煙雨 阳澄湖大闸蟹公4.5两 母3.5两 4对8只 鲜活生鲜螃蟹现货水产礼盒海鲜水',
    price: '388',
    vipPrice: '378',
    shopDesc: '自营',
    delivery: '厂商配送',
    shopName: '阳澄湖大闸蟹自营店>',
  }
  return (
    <Card
      imgUrl = { state.imgUrl }
  title = { state.title }
  price = { state.price }
  vipPrice = { state.vipPrice }
  shopDesc = { state.shopDesc }
  delivery = { state.delivery }
  shopName = { state.shopName }
    > </Card>
)
  ;
};
export default App;
```

:::

### 自定义商品标签

:::demo

```ts
import React from "react";
import { Card, Price, Tag } from '@nutui/nutui-react';

const App = () => {
  const state = {
    imgUrl:
      '//img10.360buyimg.com/n2/s240x240_jfs/t1/210890/22/4728/163829/6163a590Eb7c6f4b5/6390526d49791cb9.jpg!q70.jpg',
    title: '【活蟹】湖塘煙雨 阳澄湖大闸蟹公4.5两 母3.5两 4对8只 鲜活生鲜螃蟹现货水产礼盒海鲜水',
    price: '388',
    vipPrice: '378',
    shopDesc: '自营',
    delivery: '厂商配送',
    shopName: '阳澄湖大闸蟹自营店>',
  }
  const wordStyles = {
    padding: '0 5px',
    borderRadius: '1px',
    fontSize: '10px',
    height: '15px',
    lineHeight: '15px',
    color: '#999',
    backgroundColor: '#f2f2f7',
    marginRight: '5px',
  }
  return (
    <Card
      imgUrl = { state.imgUrl }
  title = { state.title }
  price = { state.price }
  vipPrice = { state.vipPrice }
  shopDesc = { state.shopDesc }
  delivery = { state.delivery }
  shopName = { state.shopName }
  prolistTpl = {
    < div
  className = "search_prolist_attr"
  style = {
  {
    display: 'inline-flex', margin
  :
    '3px 0 1px', height
  :
    '15px'
  }
}
>
  {
    ['鲜活', '礼盒', '国产'].map((item) => {
      return (
        <span style = { wordStyles }
      className = "word"
      key = { item } >
        { item }
        < /span>
    )
    })
  }
  </div>
}
>
  </Card>
)
  ;
};
export default App;

```

:::

### 价格后自定义标签

:::demo

```ts
import React from "react";
import { Card, Price, Tag } from '@nutui/nutui-react';

const App = () => {
  const state = {
    imgUrl:
      '//img10.360buyimg.com/n2/s240x240_jfs/t1/210890/22/4728/163829/6163a590Eb7c6f4b5/6390526d49791cb9.jpg!q70.jpg',
    title: '【活蟹】湖塘煙雨 阳澄湖大闸蟹公4.5两 母3.5两 4对8只 鲜活生鲜螃蟹现货水产礼盒海鲜水',
    price: '388',
    vipPrice: '378',
    shopDesc: '自营',
    delivery: '厂商配送',
    shopName: '阳澄湖大闸蟹自营店>',
  }
  const tagStyles = {
    display: 'inline-block',
    verticalAlign: 'middle',
    marginRight: '5px',
    marginLeft: '2px',
    height: '14px',
  }
  return (
    <Card
      imgUrl = { state.imgUrl }
  title = { state.title }
  price = { state.price }
  vipPrice = { state.vipPrice }
  shopDesc = { state.shopDesc }
  delivery = { state.delivery }
  shopName = { state.shopName }
  originTpl = {
    < img
  style = { tagStyles }
  src = "https://img11.360buyimg.com/jdphoto/s58x28_jfs/t9451/359/415622649/15318/b0943e5d/59a78495N3bd2a9f8.png"
  alt = ""
    / >
}
>
  </Card>
)
  ;
};
export default App;
```

:::

### 自定义店铺介绍

:::demo

```ts
import React from "react";
import { Card, Price, Tag } from '@nutui/nutui-react';

const App = () => {
  const state = {
    imgUrl:
      '//img10.360buyimg.com/n2/s240x240_jfs/t1/210890/22/4728/163829/6163a590Eb7c6f4b5/6390526d49791cb9.jpg!q70.jpg',
    title: '【活蟹】湖塘煙雨 阳澄湖大闸蟹公4.5两 母3.5两 4对8只 鲜活生鲜螃蟹现货水产礼盒海鲜水',
    price: '388',
    vipPrice: '378',
    shopDesc: '自营',
    delivery: '厂商配送',
    shopName: '阳澄湖大闸蟹自营店>',
  }
  return (
    <Card
      imgUrl = { state.imgUrl }
  title = { state.title }
  price = { state.price }
  vipPrice = { state.vipPrice }
  shopDesc = { state.shopDesc }
  delivery = { state.delivery }
  shopName = { state.shopName }
  shopTagTpl = { < div > 自定义店铺介绍 < /div>}
    > </Card>
)
  ;
}
  ;
  export default App;
```

:::

### 自定义右下角内容

:::demo

```ts
import React from "react";
import { Card, Price, Tag } from '@nutui/nutui-react';

const App = () => {
  const state = {
    imgUrl:
      '//img10.360buyimg.com/n2/s240x240_jfs/t1/210890/22/4728/163829/6163a590Eb7c6f4b5/6390526d49791cb9.jpg!q70.jpg',
    title: '【活蟹】湖塘煙雨 阳澄湖大闸蟹公4.5两 母3.5两 4对8只 鲜活生鲜螃蟹现货水产礼盒海鲜水',
    price: '388',
    vipPrice: '378',
    shopDesc: '自营',
    delivery: '厂商配送',
    shopName: '阳澄湖大闸蟹自营店>',
  }

  return (
    <Card
      imgUrl = { state.imgUrl }
  title = { state.title }
  price = { state.price }
  vipPrice = { state.vipPrice }
  shopDesc = { state.shopDesc }
  delivery = { state.delivery }
  shopName = { state.shopName }
  footerTpl = { < div
  style = {
  {
    fontSize: '12px'
  }
}>
  自定义 < /div>}
  > </Card>
)
  ;
}
  ;
  export default App;
```

:::

## Card

### Props

| 属性 | 说明                                       | 类型    | 默认值    |
|---------|--------------------------------------------|---------|-----------|
| imgUrl   | 左侧图片Url                                 | string  | -         |
| title     | 标题                   | string  | -    |
| price | 商品价格                         | string  | -      |
| vipPrice     | 会员价格                               | string | -    |
| shopDesc  | 店铺介绍                                  | string | -    |
| delivery     | 配送方式 | string  | -      |
| shopName   | 店铺名称| string  | -      |
| prolistTpl   | 自定义商品介绍| ReactNode  | -      |
| originTpl   | 价格后方自定义内容| ReactNode  | -      |
| shopTagTpl   | 店铺介绍自定义| ReactNode  | -      |
| footerTpl   | 右下角内容自定义| ReactNode  | -      |

## 主题定制

### 样式变量

组件提供了下列 CSS 变量，可用于自定义样式，使用方法请参考 [ConfigProvider 组件](#/zh-CN/component/configprovider)。

| 名称 |  说明 |默认值 |
| --- | --- | --- |
| --nutui-card-font-size-0 |卡片标签的字体大小 | `$font-size-0` |
| --nutui-card-left-border-radius | 卡片图片的圆角大小 | `0` |
| --nutui-card-left-background-color |  卡片图片的背景颜色 |`inherit` |
