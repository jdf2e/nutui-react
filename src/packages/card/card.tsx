import React, { FunctionComponent, CSSProperties } from 'react'
import './card.scss'
import Price from '@/packages/price'
import Tag from '@/packages/tag'

import bem from '@/utils/bem'

export interface CardProps {
  imgUrl: string
  title: string
  price: string
  vipPrice: string
  shopDesc: string
  delivery: string
  shopName: string
  className: string
  style: React.CSSProperties
  prolistTpl: React.ReactChild
  shopTagTpl: React.ReactChild
  originTpl: React.ReactChild
  footerTpl: React.ReactChild
}
const defaultProps = {} as CardProps
export const Card: FunctionComponent<Partial<CardProps> & React.HTMLAttributes<HTMLDivElement>> = (
  props
) => {
  const {
    className,
    style,
    imgUrl,
    title,
    price,
    vipPrice,
    shopDesc,
    delivery,
    shopName,
    shopTagTpl,
    originTpl,
    prolistTpl,
    footerTpl,
    ...rest
  } = {
    ...defaultProps,
    ...props,
  }
  const b = bem('card')
  return (
    <div className={`${b()} ${className}`} style={style} {...rest}>
      <div className={b('left')}>
        <img src={imgUrl} alt="" />
      </div>
      <div className={b('right')}>
        <div className={b('right__title')}>{title}</div>
        {prolistTpl}
        <div className={b('right__price')}>
          <Price price={price}></Price>
          {originTpl || <Price className={b('right__price__origin')} price={vipPrice}></Price>}
        </div>
        <div className={b('right__other')}>
          {shopTagTpl || (
            <>
              <Tag type="danger">{shopDesc}</Tag>
              <Tag plain>{delivery}</Tag>
            </>
          )}
        </div>
        <div className={b('right__shop')}>
          <div className={b('right__shop__name')}>{shopName}</div>
          {footerTpl}
        </div>
      </div>
    </div>
  )
}

Card.defaultProps = defaultProps
Card.displayName = 'NutCard'
