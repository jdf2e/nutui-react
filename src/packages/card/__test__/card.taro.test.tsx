import TestUtils from '@tarojs/test-utils-react'
import '@testing-library/jest-dom'
import { Card } from '../card.taro'

const testUtils = new TestUtils()
const execFn = (desc: string, component: any, props: object, cb: any) => {
  it(desc, async () => {
    await testUtils.mount(component, {
      props,
    })
    cb()
    testUtils.unmout()
  })
}
const state = {
  src: '//img10.360buyimg.com/n2/s240x240_jfs/t1/210890/22/4728/163829/6163a590Eb7c6f4b5/6390526d49791cb9.jpg!q70.jpg',
  title:
    '活蟹】湖塘煙雨 阳澄湖大闸蟹公4.5两 母3.5两 4对8只 鲜活生鲜螃蟹现货水产礼盒海鲜水',
  price: '388',
  vipPrice: '378',
  shopDescription: '自营',
  delivery: '厂商配送',
  shopName: '阳澄湖大闸蟹自营店>',
}
describe('card-taro', () => {
  execFn(
    'props test',
    Card,
    {
      src: state.src,
      title: state.title,
      price: state.price,
      vipPrice: state.vipPrice,
      shopDescription: state.shopDescription,
      delivery: state.delivery,
      shopName: state.shopName,
    },
    async () => {
      const priceDoms = testUtils.queries.querySelectorAll(
        '.nut-price-integer-normal'
      )
      const tagDoms = testUtils.queries.querySelectorAll('.nut-tag')
      expect(
        testUtils.queries
          .querySelector('.nut-card-left img')
          ?.getAttribute('src')
      ).toBe(state.src)
      expect(
        testUtils.queries.querySelector('.nut-card-right-title')
      ).toContainHTML(state.title)
      expect(priceDoms[0].innerHTML).toBe(state.price)
      expect(priceDoms[1].innerHTML).toBe(state.vipPrice)
      expect(tagDoms[0]).toContainHTML(
        `<div class="nut-tag nut-tag-danger"><span class="nut-tag-text">${state.shopDescription}</span></div>`
      )
      expect(tagDoms[1]).toContainHTML(
        `<div class="nut-tag nut-tag-default nut-tag-plain"><span class="nut-tag-text">${state.delivery}</span></div>`
      )
      expect(
        testUtils.queries.querySelector('.nut-card-right-shop-name')
      ).toContainHTML(state.shopName)
      expect(testUtils.html()).toMatchSnapshot()
    }
  )
})
