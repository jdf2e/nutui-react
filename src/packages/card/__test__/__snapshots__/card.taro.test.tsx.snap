// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card-taro props test 1`] = `
"[36m<body>[39m
  [36m<div>[39m
    [36m<div[39m
      [33mclass[39m=[32m"nut-card"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m"nut-card-left"[39m
      [36m>[39m
        [36m<img[39m
          [33malt[39m=[32m""[39m
          [33msrc[39m=[32m"//img10.360buyimg.com/n2/s240x240_jfs/t1/210890/22/4728/163829/6163a590Eb7c6f4b5/6390526d49791cb9.jpg!q70.jpg"[39m
        [36m/>[39m
      [36m</div>[39m
      [36m<div[39m
        [33mclass[39m=[32m"nut-card-right"[39m
      [36m>[39m
        [36m<div[39m
          [33mclass[39m=[32m"nut-card-right-title"[39m
        [36m>[39m
          [0m活蟹】湖塘煙雨 阳澄湖大闸蟹公4.5两 母3.5两 4对8只 鲜活生鲜螃蟹现货水产礼盒海鲜水[0m
        [36m</div>[39m
        [36m<div[39m
          [33mclass[39m=[32m"nut-card-right-price"[39m
        [36m>[39m
          [36m<div[39m
            [33mclass[39m=[32m"nut-price  "[39m
          [36m>[39m
            [36m<div[39m
              [33mclass[39m=[32m"nut-price-symbol nut-price-symbol-normal"[39m
            [36m>[39m
              [0m￥[0m
            [36m</div>[39m
            [36m<div[39m
              [33mclass[39m=[32m"nut-price-integer nut-price-integer-normal"[39m
            [36m>[39m
              [0m388[0m
            [36m</div>[39m
            [36m<div[39m
              [33mclass[39m=[32m"nut-price-decimal nut-price-decimal-normal"[39m
            [36m>[39m
              [0m.[0m
            [36m</div>[39m
            [36m<div[39m
              [33mclass[39m=[32m"nut-price-decimal nut-price-decimal-normal"[39m
            [36m>[39m
              [0m00[0m
            [36m</div>[39m
          [36m</div>[39m
          [36m<div[39m
            [33mclass[39m=[32m"nut-price  nut-card-right-price-origin"[39m
          [36m>[39m
            [36m<div[39m
              [33mclass[39m=[32m"nut-price-symbol nut-price-symbol-normal"[39m
            [36m>[39m
              [0m￥[0m
            [36m</div>[39m
            [36m<div[39m
              [33mclass[39m=[32m"nut-price-integer nut-price-integer-normal"[39m
            [36m>[39m
              [0m378[0m
            [36m</div>[39m
            [36m<div[39m
              [33mclass[39m=[32m"nut-price-decimal nut-price-decimal-normal"[39m
            [36m>[39m
              [0m.[0m
            [36m</div>[39m
            [36m<div[39m
              [33mclass[39m=[32m"nut-price-decimal nut-price-decimal-normal"[39m
            [36m>[39m
              [0m00[0m
            [36m</div>[39m
          [36m</div>[39m
        [36m</div>[39m
        [36m<div[39m
          [33mclass[39m=[32m"nut-card-right-other"[39m
        [36m>[39m
          [36m<div[39m
            [33mclass[39m=[32m"nut-tag nut-tag-danger"[39m
          [36m>[39m
            [36m<span[39m
              [33mclass[39m=[32m"nut-tag-text"[39m
            [36m>[39m
              [0m自营[0m
            [36m</span>[39m
          [36m</div>[39m
          [36m<div[39m
            [33mclass[39m=[32m"nut-tag nut-tag-default nut-tag-plain"[39m
          [36m>[39m
            [36m<span[39m
              [33mclass[39m=[32m"nut-tag-text"[39m
            [36m>[39m
              [0m厂商配送[0m
            [36m</span>[39m
          [36m</div>[39m
        [36m</div>[39m
        [36m<div[39m
          [33mclass[39m=[32m"nut-card-right-shop"[39m
        [36m>[39m
          [36m<div[39m
            [33mclass[39m=[32m"nut-card-right-shop-name"[39m
          [36m>[39m
            [0m阳澄湖大闸蟹自营店&gt;[0m
          [36m</div>[39m
        [36m</div>[39m
      [36m</div>[39m
    [36m</div>[39m
  [36m</div>[39m
[36m</body>[39m"
`;
